internal/impl/devbox.go

```diff
@@ -243,7 +235,7 @@ func (d *Devbox) Shell(ctx context.Context) error {
 		WithShellStartTime(shellStartTime),
 	}
 
-	shell, err := NewDevboxShell(d.cfg.Nixpkgs.Commit, opts...)
+	shell, err := NewDevboxShell(d.cfg.NixPkgsCommitHash(), opts...)
```

Comment:
similarly, here, we need to change `nix.FlakeNixpkgs` to work with empty commit hashes